README.md

```diff
@@ -894,239 +912,207 @@ How to test actions with scenario:
+# have the same destination.
+state = scenario.State(stored_state={
+    "MyCharm.mystored_state": scenario.StoredState({
+        'foo': 'bar',
+        'baz': {42: 42},
+    })
+])
 ```
 
 And the charm's runtime will see `self.stored_State.foo` and `.baz` as expected. Also, you can run assertions on it on
 the output side the same as any other bit of state.
 
-# Resources
+## Resources
+
+TONY NOTE: this belongs up in the state section
+TONY NOTE: ideally there's a way to say "resource not attached", maybe if the path is not provided/is None?
 
 If your charm requires access to resources, you can make them available to it through `State.resources`.
-From the perspective of a 'real' deployed charm, if your charm _has_ resources defined in `metadata.yaml`, they _must_ be made available to the charm. That is a Juju-enforced constraint: you can't deploy a charm without attaching all resources it needs to it.
-However, when testing, this constraint is unnecessarily strict (and it would also mean the great majority of all existing tests would break) since a charm will only notice that a resource is not available when it explicitly asks for it, which not many charms do.
+From the perspective of a 'real' deployed charm, if your charm _has_ oci-image resources defined in `metadata.yaml`, they _must_ be made available to the charm. That is a Juju-enforced constraint: you can't deploy a charm without attaching all oci-image resources it needs to it. However, you can deploy a charm with missing file resources, although doing this
+will lead to errors when your charm tries to access those resources.
 
-So, the only consistency-level check we enforce in Scenario when it comes to resource is that if a resource is provided in State, it needs to have been declared in metadata.
+So, the only consistency-level check we enforce in Scenario when it comes to resources is that if a resource is provided in State, it needs to have been declared in metadata.
 
 ```python
-from scenario import State, Context
-ctx = Context(MyCharm, meta={'name': 'juliette', "resources": {"foo": {"type": "oci-image"}}})
-with ctx.manager("start", State(resources={'foo': '/path/to/resource.tar'})) as mgr:
-    # if the charm, at runtime, were to call self.model.resources.fetch("foo"), it would get '/path/to/resource.tar' back.
-    path = mgr.charm.model.resources.fetch('foo')
-    assert path == '/path/to/resource.tar' 
+import ops
+import scenario
+
+class MyCharm(ops.CharmBase):
+    def __init__(self, framework):
+        super().__init__(framework)
+        framework.observe(self.on.start, self._on_start)
+
+    def _on_start(self, _):
+        path = self.model.resources.fetch('foo')  # path == pathlib.Path('/path/to/resource.tar')
+
+
+ctx = scenario.Context(MyCharm, meta={'name': 'juliette', "resources": {"foo": {"type": "oci-image"}}})
+# Again, the list or dict or set here is tricky, wanting to avoid duplication, but the object needing
+# to know its own name.
```

Comment:
Again, does the object need to know it's own name, and if so, can we have `run` or `State()` add it?